storage/snapshot/iterator.rs

```diff
@@ -217,8 +222,8 @@ impl LendingIterator for SnapshotRangeIterator {
         }
         match self.ready_item_source.take() {
             Some(ReadyItemSource::Both) => {
```

Comment:
If we have the item both in buffer and storage, we need to get the buffered one!